Normally kdebugdialog enables all debugging information which isn't
explicitly turned off. This patch reverses this behaviour. Only the 
debugging areas which are enabled are checked.
See also kdelibs' 09_disable_debug_messages_if_not_explicitly_enabled.diff
which does the same when printing debug information. Don't print them
unless the user requested them.
--- a/kdebugdialog/klistdebugdialog.cpp
+++ b/kdebugdialog/klistdebugdialog.cpp
@@ -124,7 +124,7 @@
     QListWidgetItem* item = m_areaWidget->item(i);
     KConfigGroup group = pConfig->group( item->data(Qt::UserRole).toByteArray() ); // Group name = debug area code = cb's name
 
-    int setting = group.readEntry( "InfoOutput", 2 );
+    int setting = group.readEntry( "InfoOutput", 4 );
 
     switch (setting) {
       case 4: // off
